test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -1739,23 +1762,186 @@ public void SelectColumnsOnnxTest()
+        {
+            // By default when exporting a ML.NET model to Onnx,
+            // the onnx model will contain an "input node" for every
+            // column in the input schema of the input dataview.
+            // "Output nodes" for each one of the columns in the output
+            // schema are also added. The input columns that were originally
+            // propagated to the output, also get a connection in the onnx model
+            // to propagate that information from each input node to the corresponding
+            // output node.
+            //
+            // Only when using the ColumnSelectingTransformer is it possible to drop
+            // input columns to make them dissapear from the output. Since they dissapear from
+            // the output schema, the exported onnx model doesn't contain output nodes for the
+            // dropped columns. The OnnxTransformer shouldn't propagate this input columns to the output
+            // but it should drop them as well.
+            //
+            // On the other hand, OnnxTransformer should propagate all the columns that are in its input schema,
+            // but which are not mentioned in the onnx model (particullarly not mentioned in the input nodes),
+            // because this would mean that the onnx model should simply ignore them, without dropping them.
+            //
+            // In general, to support the above behavior the following cases were agreed, and tested in here:
+            //
+            // Case 1. An input column that has a corresponding input node in the onnx model, but not connected to a corresponding output node,
+            //  should be dropped and not propagated to the output schema. This case can only happen if the input column was dropped in the pipeline
+            //  that was exported (case 1a), or if another column with the same name is created in the pipeline (case 1b).
+            // Case 2. An input column that has a corresponding input node in the onnx model directly connected to an output node with the same name,
+            //  should be propagated from the input to the output schema. This is the usual behavior unless a column is dropped or hidden in the
+            //  original pipeline (which then would be case 1).
+            // Case 3. A column that was created and dropped inside the pipeline that was exported to onnx, should be dropped.
+            // Case 4. A column that was created but not dropped inside the pipeline that was exported to onnx, should not be dropped
+            // Case 5. An input column that doesn't have a corresponding input node in the onnx model (i.e., that isn't even mentioned
+            //  by the model) should be propagated as if being untouched.
+            // Case 6. An input column that doesn't have a corresponding input node in the onnx model, but that has the same name as an output
+            //  column created by the onnx transformer, should be hidden.
+            //
```

Comment:
I need to add case 7: a column that was created and hidden inside the pipeline being exported. It will be dropped.

Also perhaps consider adding to the test the case where and input column that will be dropped gets copied into another column of another name. This will technically be case 4, but it might still be good to include it.